source/SharpGL/Core/SharpGL.WPF/OpenGLControl.xaml.cs

```diff
@@ -31,21 +23,66 @@ public partial class OpenGLControl : UserControl
         public OpenGLControl()
         {
             InitializeComponent();
-            SizeChanged += new SizeChangedEventHandler(OpenGLControl_SizeChanged);
+
+            timer = new DispatcherTimer();
+
+            Unloaded += OpenGLControl_Unloaded;
+            Loaded += OpenGLControl_Loaded;
+        }
+
+        /// <summary>
+        /// Handles the Loaded event of the OpenGLControl control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="System.Windows.RoutedEventArgs"/> Instance containing the event data.</param>
+        private void OpenGLControl_Loaded(object sender, RoutedEventArgs routedEventArgs)
+        {
+            SizeChanged += OpenGLControl_SizeChanged;
+
+            timer.Tick -= timer_Tick;
+            timer.Tick += timer_Tick; // to avoid twice call, ensure timer_Tick event handler is registered only once, because it is also registred on apply template
+
+            UpdateOpenGLControl((int) RenderSize.Width, (int) RenderSize.Height);
+
+            //  DispatcherTimer setup
+            timer.Tick += new EventHandler(timer_Tick);

```

Comment:
both lines 42 and 43 can go, there should be no way to have the event handler registered twice so no need to deregister it on `Loaded`